### ISSUE TITLE
Error when using `upgrade_config_file.py` on new dataset yamls in riboviz 2.0

### ISSUE THREAD

**Issue by FlicAnderson** [riboviz/riboviz#203]
Hi folks,

I've run into an issue when I've tried to run the upgrade config script on the Cryptococcus yaml files from [example-datasets repo Cryptococcus branch](https://github.com/riboviz/example-datasets/tree/CryptococcusH99-3/fungi/cryptococcus).

I had cloned example-datasets into home (`~/example-datasets`), checked out the `CryptococcusH99-3` branch & the yamls I wanted to upgrade were at `~/example-datasets/fungi/cryptococcus/`.  I've tried the H99 .yaml first.

From the `~/riboviz` folder: 
I ran: 
`python -m riboviz.tools.upgrade_config_file -i ~/example-datasets/fungi/cryptococcus/Wallace_2020_H99_4-samples_10p_up12dwn9_CDS_120bpL_120bpR_config.yaml -o ~/example-datasets/fungi/cryptococcus/Wallace_2020_H99_4-samples_10p_up12dwn9_CDS_120bpL_120bpR_config_upgraded.yaml`
```
(base) flic@circe:~/riboviz$ python -m riboviz.tools.upgrade_config_file -i ~/example-datasets/fungi/cryptococcus/Wallace_2020_H99_4-samples_10p_up12dwn9_CDS_120bpL_120bpR_config.yaml -o ~/example-datasets/fungi/cryptococcus/Wallace_2020_H99_4-samples_10p_up12dwn9_CDS_120bpL_120bpR_config_upgraded.yaml 
Traceback (most recent call last):
  File "/home/flic/miniconda3/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/flic/miniconda3/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/flic/riboviz/riboviz/tools/upgrade_config_file.py", line 60, in <module>
    invoke_upgrade_config_file()
  File "/home/flic/riboviz/riboviz/tools/upgrade_config_file.py", line 56, in invoke_upgrade_config_file
    upgrade_config.upgrade_config_file(input_file, output_file)
  File "/home/flic/riboviz/riboviz/upgrade_config.py", line 186, in upgrade_config_file
    upgrade_config(config)
  File "/home/flic/riboviz/riboviz/upgrade_config.py", line 161, in upgrade_config
    features_dir_file = os.path.split(features_file)
  File "/home/flic/miniconda3/lib/python3.7/posixpath.py", line 107, in split
    p = os.fspath(p)
TypeError: expected str, bytes or os.PathLike object, not NoneType
```

I'm not sure whether this is an issue with version 2.0 parameters being slightly different to the older upgrades noted in [here, upgrade_config.py](https://github.com/riboviz/riboviz/blob/master/riboviz/upgrade_config.py).

Another thing I wasn't sure about was whether the provenance header (brought in from #166 ) might not be handled correctly in the upgrade script now, since this is a new feature.  I tried running the script again after deleting these rows temporarily, but it gave the same error, so perhaps this section is ignored, but if the upgrade script was working, would these rows be left at the top of the file or reaarranged within the other parameters?

Checking through other issues, I noticed #172 which mentions a config checker, and @mikej888 seems to be working on a checker in example-datasets repo, so this may be duplicating issues @mikej888 has already picked up on in [riboviz/example-datasets/#9](https://github.com/riboviz/example-datasets/issues/9)?  This might make working on `riboviz.tools.upgrade_config_file` less important?

**Comment by mikej888** [comment]
The code should be updated to handle updates to optional parameters. The updates to the relative paths in the vignette should not be done - these paths can be updated manually. [docs/user/upgrade-1x.md](https://github.com/riboviz/riboviz/blob/master/) recommends that users check any file paths anyway.

**Comment by mikej888** [comment]
@FlicAnderson please try fix in branch optional-null-values-202-203.

**Comment by FlicAnderson** [comment]
Hi @mikej888 , branch works and upgrades the yaml!  Running nextflow --validate only validates it OK :+1: 

It does rearrange the `provenance` section alphabetically into the parameters (though it keeps the block together), but that's that won't affect how it runs and can just be shuffled about manually. 

Is there any way to make sure that if there is NO `asite_disp_length_file`, `codon_positions_file` and `t_rna_file` in the original yaml that these are set to `null` by default maybe, instead of set to the `data/[filename]` value? That might be more useful if most people aren't using _Saccharomyces cerevisiae_ datasets? But equally we could just make it clear in the documentation that these should be removed or set to null if that's the case? @ewallace what do you think would be more universally useful?

**Comment by mikej888** [comment]
Merged into "develop" branch.

**Comment by FlicAnderson** [comment]
Linking this fixed issue to example datasets issue [#92](https://github.com/riboviz/example-datasets/issues/92)
